### PR TITLE
Acceptance tests - change default path

### LINKED ISSUE #806: Stop using the default state path in the zebrad acceptance tests
When I run the zebrad command acceptance tests, they create a sled state directory at `~/.cache/zebra/mainnet/state`.

`~/.cache/...` is the default state directory path. If we use the default path, the tests could fail if there is a running zebrad instance, or if the state is from a different version of `zebrad`.

Instead, we should create a new state in a temporary directory.

Here's how we create a temporary directory in the state tests:
https://github.com/ZcashFoundation/zebra/blob/main/zebra-state/tests/basic.rs#L71

And here's the env var for changing the state directory:
https://github.com/ZcashFoundation/zebra/blob/main/zebra-state/src/lib.rs#L73

### PR DESCRIPTION
**Pull request by oxarbitrage**
This should close https://github.com/ZcashFoundation/zebra/issues/806

### CODE REVIEW

File: zebrad/tests/acceptance.rs [hunk 1]
```diff
@@ -47,6 +50,13 @@ pub fn get_child_no_args() -> Result<(zebra_test::command::TestChild, impl Drop)
     ))
 }
 
+pub fn change_default_path() -> Result<()> {
+    let tmp = TempDir::new("zebrad_tests")?;
+    std::env::set_var("ZEBRAD_CACHE_DIR", tmp.path());
+
+    Ok(())
```
Comment (teor2345) [inline zebrad/tests/acceptance.rs]: Issue: This temporary directory is deleted when this function returns, and `tmp` is dropped:
https://docs.rs/tempdir/0.3.7/tempdir/struct.TempDir.html

Deleting the directory before the test finishes could cause weird test failures, particularly on Windows, which is more strict about file and directory access.

Instead, use a subdirectory of the existing temporary directory in `test_cmd`:
https://github.com/ZcashFoundation/zebra/blob/f2d7bb3177d5df59d26746d68ae354986372f3a7/zebra-test/src/command.rs#L10
This temporary directory is kept around by the `dir` and `_guard` variables, until the test finishes.

It's ok to always set `ZEBRAD_CACHE_DIR` in `test_cmd`.

**Review by teor2345 — CHANGES_REQUESTED** [review]
Thanks for doing this fix!

Let's move it to `test_cmd`, so that it uses the existing temporary directory.

File: zebrad/tests/acceptance.rs (hunk 1 above)
Comment (oxarbitrage) [reply zebrad/tests/acceptance.rs]: Thanks and sorry for not checking the temp dir was going to be dropped. Made the suggested changes at https://github.com/ZcashFoundation/zebra/pull/808/commits/a7142d9160fa8ad89d6e1d46974ae454885ecc66

**Review by teor2345 — APPROVED** [review]
Looks good!

Let's wait for CI, then squash and merge.

**Review by dconnolly — APPROVED** [review]

**Pull request merged**